commands/trends_command.py

```diff
@@ -0,0 +1,219 @@
+import matplotlib
+import requests
+import pandas as pd
+from pytrends.request import TrendReq
+from datetime import date, timedelta
+
+from .botcommand import BotCommand
+from boterror import BotError
+from botquery import BotQuery
+from config import BASE_URL
+from constants import SEND_PHOTO_ENDPOINT, PHOTO_KEY, CHAT_ID_KEY, NEW_LINE_CHAR, SPACE, UNDERSCORE, PARSE_MODE_KEY, UNDERSCORE, SPACE
+
+"""
+returns the associated response dictionary key w.r.t to a country's name
+eg. United States -> united_states
+    canada -> Canada
+"""
+def get_country_key(name: str):
+    name_formatted = name.strip().replace(SPACE, UNDERSCORE).lower()
+    if name_formatted == 'new_zealand':
+        return 'new zealand' # For some reason, New Zealand is the only country without an underscore in the response
+    if name_formatted in { 'us', 'usa', 'states', 'america' }: # For user convenience
+        return 'united_states'
+    return name_formatted
+
+"""
+Calculates the day time_delta units away from the given date.
+time_delta_str should be in the format '<number><unit>' where number is a positive integer and unit is one of d, w, m or y
+"""
+def get_initial_date(current_date: date, time_delta_str: str):
+    allowed_units_map = {
+        'd': 1,
+        'w': 7,
+        'm': 30,
+        'y': 365
+    }
+
+    try:
+        unit = time_delta_str[-1]
+        time = int(time_delta_str[:-1])
+        
+        return current_date - timedelta(days=time * allowed_units_map[unit])
+    except:
+        raise BotError(f"Invalid argument '{time_delta_str}' for option '{TrendsCommand.TIME_FLAG}'")
+
+"""
+Validates the string 'year' is a valid year.
+"""
+def validate_year_str(year_str: str):
+    try:
+        year = int(year_str)
+        if year <= 0:
+            raise
+    except:
+        raise BotError(f"Invalid year '{year_str}'")
```

Comment:
Redundant code, please change to:
```
def validate_year_str(year_str: str):
  year = int(year_str)
  if year <= 0:
    raise BotError(f"Invalid year '{year_str}'")
```

Reply:
I have a good workaround using python's `isnumeric` string method. It should make it a lot cleaner